mautic/mautic
# Fix #9591 by filling default labels on boolean fields from the API

Whenc creating a boolean field the UI provides default values for Yes/No labels
but the API doesn't and no error were given until user navigate to a page that load lead fields.

This commit mimics the UI behavior on the API by filling default Yes and No labels automatically

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | None... 
| Related developer documentation PR URL | https://developer.mautic.org/#create-field
| Issue(s) addressed                     | Fixes #9591



#### Description:

When creating a boolean custom field in the UI, the Yes/No labels are rendered with "Yes" and "No" default values and sent to backend. But on when creating the same from the API, if no `..., properties: { "yes": "Yes", "no: "No" } ... }` is provided the field *is* created, by with blank "yes" and "no" properties. This then crashes some pages when ListBundle is loaded with classic Undefined index error.



#### Steps to test this PR:

* Enable API on Settings
* Run this curl (edit url, user and password)
  ```
  curl -v -L \
    -H 'Content-Type: application/json' \
    -u admin:xxxxxxxxx \
    http://127.0.0.1:8081/index_dev.php/api/fields/contact/new \
    -d '{ "label": "foo", "alias": "foo", "type": "boolean" }'
  ```
* Navigate to Segments and try to create a new segment, you should get an alert, 
  <img width="294" alt="image" src="https://user-images.githubusercontent.com/392820/113522454-e96f5800-9576-11eb-888b-b73913b8e40b.png">
